src/utils/fetchAndCache.js

```diff
@@ -0,0 +1,40 @@
+/**
+ * Fetch and cache a request
+ * @param {string|URL} url
+ * @param {number} maxAge max cache duration in ms
+ */
+export default async function fetchAndCache(url, maxAge = 10 * 60 * 1000) {
+  if (!('caches' in window)) {
+    return fetch(url);
+  }
+
+  let cache;
+  try {
+    const cacheName = new URL(url).origin;
+    cache = await caches.open(cacheName);
+    const response = await cache.match(url);
+    if (response && new Date(response.headers.get('Expires')) > new Date()) {
+      return response;
+    }
+  } catch (err) {
+    // eslint-disable-next-line no-console
+    console.error('Failed to use Cache API.', err);
+  }
+
+  // otherwise fetch new data and cache
+  const response = await fetch(url);
+
+  // cache response
+  if (cache && response.ok) {
+    const clonedResponse = response.clone();
```

Comment:
What does cloning do?

Reply:
We can consume this response only once. If we access its headers/content here for caching, we can't reuse them with the returned value. So, we create a copy of this response object and store that instead.